packages/editor/src/components/post-taxonomies/flat-term-selector.js

```diff
@@ -165,11 +137,22 @@ export function FlatTermSelector( { slug } ) {
 	}, [ searchResults ] );
 
 	const { editPost } = useDispatch( editorStore );
+	const { saveEntityRecord } = useDispatch( coreStore );
 
 	if ( ! hasAssignAction ) {
 		return null;
 	}
 
+	/**
+	 * Create a new term
+	 *
+	 * @param {Object} term Term object.
+	 * @return {Promise} A promise that resolves to save term object.
+	 */
+	const createTerm = ( term ) => {
+		return saveEntityRecord( 'taxonomy', slug, term );
+	};
+
```

Comment:
@fabiankaegy, we'll need to implement similar error handling for `term_exists` like the old method.

This is a rough example, which I've not tested :)

```js
async function createTerm( term ) {
	try {
		const newTerm = saveEntityRecord( 'taxonomy', slug, term, {
			throwOnError: true,
		} );
		return unescapeTerm( newTerm );
	} catch ( error ) {
		if ( error.code !== 'term_exists' ) {
			throw error;
		}

		return {
			id: error.data.term_id,
			name: term.name,
		};
	}
}
```

Reply:
@Mamaduka Just so I can properly test this, that case is for the instances where another tab / window / user creates the new term whilst you are already in the editor and therefore you try and create the same term twice?

Reply:
@fabiankaegy, that's probably a rare case.

Users might type the tag name and hit enter or paste a comma-separated list of existing tags. The editor still tries to assign these tags correctly.

## Steps to reproduce

1. Create a tag - Example
2. Open a post and then paste the following string in the flat term selector input field - `Example,Testing`
3. Compare results on the trunk and this branch.
4. Both terms will be assigned to the post on the trunk, but only `Testing` on this branch.

Reply:
Yeah, that works as advertised :) Thanks for jumping in!